### DISCUSSION TITLE
SIGSEGV on Android (CMake)

### DISCUSSION THREAD

**Discussion by kdg3737** [libusb/libusb#1470] [Q&A]
Hi,

I followed the instructions to cross-compile for Android, copied the (arm64-v8a) .so to my jniLibs folder and added it to CMakeLists.txt. Compiles and launches fine, then crashes with a SIGSEGV on the very first call into libusb:

libusb_set_option(ctx, LIBUSB_OPTION_NO_DEVICE_DISCOVERY, NULL);

I also noticed the libusb_set_option function accepts a libusb_context* while in the documentation it seems to be a libusb_context**, not sure if this is relevant.

Please help.

**Comment by mcuee** [comment]
Maybe you want to provide more info (which libusb version you use and which Android distro, etc) and hopefully the people who are familiar with Android can help.

You may also want to try the PR here.
* https://github.com/libusb/libusb/pull/1164
